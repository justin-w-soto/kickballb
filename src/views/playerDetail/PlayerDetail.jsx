import React from 'react'
import { useEffect, useState } from 'react'
import { Link, useParams } from "react-router-dom"
import { getPlayerById } from "../../services/players"

export const PlayerDetail = () => {
    const{ id } = useParams()
    const[player, setPlayer] = useState(null)
   
    useEffect(() => {
        getPlayerById(id).then((res) => setPlayer(res))
    }, [id])

    if(!player) return <h2>Loading...</h2>

    return (
        <div>
              <h1>{player.name}</h1>
        <Link className='link' to={`/teams/${player.team.id}`}>
            <h3>{player.team.name}</h3>
        </Link>
        <h3>{player.team.city}, {player.teams.state}</h3>        
        <h3>{player.position}</h3>
        <Link className='link' to='/playerlist'><p>Player Information</p></Link>
        </div>
    )
}
