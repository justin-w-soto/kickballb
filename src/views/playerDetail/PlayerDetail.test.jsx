import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { PlayerDetail } from './PlayerDetail';

it('Should render a players team info on the screen', async () => {
    const {container} = render(
        <MemoryRouter>
           
                <PlayerDetail />
          
        </MemoryRouter>
    )
    screen.getByText('Loading...')
    expect(container).toMatchSnapshot();
})