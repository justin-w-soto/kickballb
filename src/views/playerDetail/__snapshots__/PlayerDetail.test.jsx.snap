// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render a players team info on the screen 1`] = `
<div>
  <h2>
    Loading...
  </h2>
</div>
`;
